invenio_records_resources/records/systemfields/pid_statuscheck.py

```diff
@@ -48,3 +50,12 @@ def __get__(self, record, owner=None):
             return self  # returns the field itself.
         pid = getattr(record, self.key)
         return pid.status in self._pid_status
+
+    def pre_dump(self, record, data, **kwargs):
+        """Called before a record is dumped in a secondary storage system."""
+        if self._dump:
```

Comment:
Does it make sense to make that a feature in the `SystemField` class or is something really specific for the `PIDStatusCheckField`?